src/Raven.Server/Documents/Replication/Senders/MigrationReplicationDocumentSender.cs

```diff
@@ -0,0 +1,115 @@
+            var counters = database.DocumentsStorage.CountersStorage.GetCountersByBucketFrom(ctx, bucket, etag + 1, caseInsensitiveCounters);
+            var timeSeries = database.DocumentsStorage.TimeSeriesStorage.GetSegmentsByBucketFrom(ctx, bucket, etag + 1);
+            var deletedTimeSeriesRanges = database.DocumentsStorage.TimeSeriesStorage.GetDeletedRangesByBucketFrom(ctx, bucket, etag + 1);
+
+            using (var docsIt = docs.GetEnumerator())
+            using (var tombsIt = tombs.GetEnumerator())
+            using (var conflictsIt = conflicts.GetEnumerator())
+            using (var versionsIt = revisions.GetEnumerator())
+            using (var attachmentsIt = attachments.GetEnumerator())
+            using (var countersIt = counters.GetEnumerator())
+            using (var timeSeriesIt = timeSeries.GetEnumerator())
+            using (var deletedTimeSeriesRangesIt = deletedTimeSeriesRanges.GetEnumerator())
+            using (var mergedInEnumerator = new MergedReplicationBatchEnumerator(stats.DocumentRead, stats.AttachmentRead, stats.TombstoneRead, stats.CounterRead, stats.TimeSeriesRead))
+            {
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.Document, docsIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.DocumentTombstone, tombsIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.Document, conflictsIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.Document, versionsIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.Attachment, attachmentsIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.CounterGroup, countersIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.TimeSeriesSegment, timeSeriesIt);
+                mergedInEnumerator.AddEnumerator(ReplicationBatchItem.ReplicationItemType.DeletedTimeSeriesRange, deletedTimeSeriesRangesIt);
+
+                while (mergedInEnumerator.MoveNext())
+                {
+                    yield return mergedInEnumerator.Current;
+                }
+            }
+        }
+        */
+
+        protected override bool ShouldSkip(ReplicationBatchItem item, OutgoingReplicationStatsScope stats, SkippedReplicationItemsInfo skippedReplicationItemsInfo)
+        {
+            switch (item)
+            {
+                case DocumentReplicationItem doc:
+                    if (doc.Flags.Contain(DocumentFlags.Artificial))
+                    {
+                        stats.RecordArtificialDocumentSkip();
+                        skippedReplicationItemsInfo.Update(item, isArtificial: true);
+                        return true;
+                    }
+
+                    if (doc.Id.StartsWith(HiLoHandler.RavenHiloIdPrefix, StringComparison.OrdinalIgnoreCase))
```

Comment:
why are you skipping hilos?

Reply:
hmm I was under the impression that the HiLo scope is per shard and not per bucket.. will remove this line.